Subject: [PATCH] fix(VarVal): fixing ambiguous overload
---
Index: nautilus/include/nautilus/val.hpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/nautilus/include/nautilus/val.hpp b/nautilus/include/nautilus/val.hpp
--- a/nautilus/include/nautilus/val.hpp	(revision 5fa4c9043d961238d283bf129b82c59e1476974a)
+++ b/nautilus/include/nautilus/val.hpp	(revision d4e4829c76299890d8fdded5291caa92300ca2fc)
@@ -350,13 +350,13 @@
 		return details::FUNC(std::forward<decltype(lhsV)>(lhsV), std::forward<RHS>(right));                            \
 	}                                                                                                                  \
                                                                                                                        \
-	template <typename LHS, typename RHS>                                                                              \
-	    requires(is_##CATEGORY##_val<LHS> && !is_##CATEGORY##_val<RHS> &&                                              \
-	             !std::is_##CATEGORY##_v<std::remove_cvref_t<RHS>> && convertible_to_##CATEGORY##_val<RHS, LHS>)       \
-	auto inline operator OP(LHS&& left, RHS&& right) {                                                                 \
-		auto rV = static_cast<std::remove_cvref_t<LHS>>(right);                                                        \
-		return left OP rV;                                                                                             \
-	}
+    template <typename LHS, typename RHS>                                                                              \
+        requires(is_##CATEGORY##_val<LHS> && !is_##CATEGORY##_val<RHS> &&                                              \
+                 !std::is_##CATEGORY##_v<std::remove_cvref_t<RHS>> && convertible_to_##CATEGORY##_val<RHS, LHS>)       \
+    auto inline operator OP(LHS&& left, RHS&& right) {                                                                 \
+       auto rV = static_cast<std::remove_cvref_t<LHS>>(right);                                                         \
+       return details::FUNC(std::forward<LHS>(left), std::forward<decltype(rV)>(rV));                                  \
+    }

 DEFINE_BINARY_OPERATOR(+, add, fundamental)

